xenomint/mux.go

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2018 The CovenantSQL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package xenomint
+
+import (
+	"sync"
+
+	"github.com/CovenantSQL/CovenantSQL/proto"
+	"github.com/CovenantSQL/CovenantSQL/rpc"
+	wt "github.com/CovenantSQL/CovenantSQL/worker/types"
+)
+
+// MuxService defines multiplexing service of xenomint chain.
+type MuxService struct {
+	ServiceName string
+	// serviceMap maps DatabaseID to *Chain instance.
+	serviceMap sync.Map
+}
+
+func NewMuxService(name string, server *rpc.Server) (service *MuxService, err error) {
+	var s = &MuxService{
+		ServiceName: name,
+	}
+	if err = server.RegisterService(name, s); err != nil {
+		return
+	}
+	service = s
+	return
+}
+
+func (s *MuxService) register(id proto.DatabaseID, c *Chain) {
+	s.serviceMap.Store(id, c)
+}
+
+func (s *MuxService) unregister(id proto.DatabaseID) {
+	s.serviceMap.Delete(id)
+}
+
+type MuxQueryRequest struct {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type MuxQueryRequest should have comment or be unexported